### ISSUE TITLE
Introducing feature branches ahead of github pages episode

### ISSUE THREAD

**Issue by elopatin-uc3** [LibraryCarpentry/lc-git#54]
One of the critical practices that enables collaboration on a project in git is feature branching. At this time, branching is first mentioned in the Github Page episode, but is not prefaced by an introduction or conceptual explanation. 

It's likely to be helpful if a visual reference to a feature branch, the purpose behind creating one, and the steps to create a branch off of master or develop are introduced ahead of the Pages episode. Creating a standalone episode about branching, before diving into Pages, could be considered as well.

An example feature branch diagram (using master, rather than develop):

![feature_branch](https://user-images.githubusercontent.com/49419424/57823597-f6012200-774c-11e9-9e19-a707d1c41c60.png)

Although discussing something like gitflow is too heavyweight, familiarizing learners with creating and merging simple feature branches will enable them to work on a small team concurrently without stepping on each other's toes.

**Comment by dcmcand** [comment]
I agree this is definitely needed. Branching is core to using git according to best practices and should be taught. We should work this in with the lesson restructure we are working on.

**Comment by emcaulay** [comment]
I just wanted to comment here and let you know that I have become a new maintainer on this lesson, and I am addressing high priority issues first. 

I hope to return to this issue in October or November 2021. It seems worth addressing!

Thank you to all contributors for raising it and getting it started.

**Comment by doingarchives** [comment]
Here is perhaps our first agile story point - we should be mindful that librarianship, archives, records management and museums are all places where our learners might have context and a sense of belonging - let's work to translate this software functionality to an example that is teachable and corrects for some of the bias and discontinuity observed  in the course.

**Comment by alex-ball** [comment]
Thanks again for raising this point for discussion.

I agree that feature branches are a valuable part of the Git toolkit, and using them is a good habit to get into to avoid breakages and conflicts in the main branch.

As far as I can see at the time of writing, both this lesson and Software Carpentry's [git-novice](https://github.com/swcarpentry/git-novice) lesson stick to working in one branch; so introducing the idea of working across multiple branches would widen the scope of the lessons, meaning more cognitive load. I think this is something we'd need advice from @LibraryCarpentry/curriculum-advisors on.

**Comment by alex-ball** [comment]
I've been looking at this again, as the new optional content in episode 5 does include switching to a new branch.

Episode 2 promises twice that branches will be covered later, but the word is just used without explanation in episodes 3 and 5.

At a minimum, I think we should have a callout box explaining branches, similar to the one explaining the staging area. Possibly the most natural place to put it would be at the end of the section "[Displaying the current project’s status](https://librarycarpentry.github.io/lc-git/02-getting-started.html#displaying-the-current-projects-status)" in episode 2, instead of saying "more on this later". But as there is a lot going on in that episode, it will need to be simple to keep the cognitive load down.
